cds: use "IsActiveEntity" in anno expressions (#1134)
cap-js/docs: cds/cdl.md

* If the annotation is assigned to a subelement of a structured element, the top level
  elements of the entity can be accessed via `$self`.
* A parameter `par` can be accessed via `:par`, just like parameters of a parametrized entity in queries.

> * For a draft-enabled entity, element `IsActiveEntity` can be accessed via prefix `$draft`.

Reviewer: would using the word 'property' instead of 'element' here be better?

Author: In CAP we usually use "element" (or maybe "field") rather than "property", so I wouldn't change it.

* If the annotation is assigned to a subelement of a structured element, the top level
  elements of the entity can be accessed via `$self`.
* A parameter `par` can be accessed via `:par`, just like parameters of a parametrized entity in queries.
* The draft specific element `IsActiveEntity` can be referred to with the magic variable `$draft.IsActiveEntity`.

>   During draft augmentation `$draft.IsActiveEntity` is rewritten to `$self.IsActiveEntity` for all draft enabled

Reviewer: ```suggestion
  During draft augmentation `$draft.IsActiveEntity` is rewritten to `$self.IsActiveEntity` for all root and sub nodes of draft enabled
```
does this rewrite mean that it's $draft.IAE in some handlers and then it's always $self.IAE?

Reviewer 2: it has nothing to do with handlers...

Reviewer: ```suggestion
 It can be reffered to by `$self.IsActiveEntity` in all root and sub nodes of draft enabled
```

Reviewer 2: no please not. `$self.IsActiveEntity` IS A REFERENCE, a reference can't be referenced.... Why is there always the urge to improve something that is already exactly what it should be?

>   entities (root and sub nodes but not for named types or entity parameters).

Reviewer: ```suggestion
  entities (but not for named types or entity parameters).
```

Reviewer 2: sorry, no in a draft context the state elements are available in both the root and the sub nodes, that's important to know.

Reviewer: I mentioned that in the previous line.

Reviewer 2: so what's the difference? In fact the content in parentheses are more like an additional information... Just leave it as it is, it's good enough

* If the annotation is assigned to a subelement of a structured element, the top level
  elements of the entity can be accessed via `$self`.
* A parameter `par` can be accessed via `:par`, just like parameters of a parametrized entity in queries.

> * The draft specific element `IsActiveEntity` can be referred to with the magic variable `$draft.IsActiveEntity`.

Author: hm, compared to the other entries in this list, this is a rather long and detailed description for a special case

Reviewer 2: yes that's right, it's a special case that (what I think) needs some more detailed explanation (all others in this list are straight forward and will always work)

Author: ok
